client/sysinfo/src/sysinfo.rs

```diff
@@ -32,14 +33,76 @@ use std::{
 	time::{Duration, Instant},
 };
 
+/// Throughput as measured in bytes per second.
+#[derive(Deserialize, Serialize, Debug, Clone, Copy, PartialEq, PartialOrd)]
+pub struct Throughput(f64);
+
+const KIBIBYTE: f64 = 1024.0;
+
+impl Throughput {
+	/// `f64` kibibyte/s to byte/s.
+	pub fn from_kibs(kibs: f64) -> Throughput {
+		Throughput(kibs * KIBIBYTE)
+	}
+
+	/// `f64` mebibyte/s to byte/s.
+	pub fn from_mibs(mibs: f64) -> Throughput {
+		Throughput(mibs * KIBIBYTE * KIBIBYTE)
+	}
+
+	/// `f64` gibibyte/s to byte/s.
+	pub fn from_gibs(gibs: f64) -> Throughput {
+		Throughput(gibs * KIBIBYTE * KIBIBYTE * KIBIBYTE)
+	}
+
+	pub fn as_f64(&self) -> f64 {
+		self.0
+	}
+
+	/// [`Self`] as number of kibibyte/s.
+	pub fn as_kibs(&self) -> f64 {
+		self.0 / KIBIBYTE
+	}
+
+	/// [`Self`] as number of mebibyte/s.
+	pub fn as_mibs(&self) -> f64 {
+		self.0 / (KIBIBYTE * KIBIBYTE)
+	}
+
+	/// [`Self`] as number of gibibyte/s.
+	pub fn as_gibs(&self) -> f64 {
+		self.0 / (KIBIBYTE * KIBIBYTE * KIBIBYTE)
+	}
+
+	/// Normalizes [`Self`] to use the larges unit possible.
+	fn normalize(&self) -> (f64, &'static str) {
```

Comment:
I don't think we need the `normalize` function to be public anymore since it is only used for formatting.

Reply:
Yeah, if it's not used outside of this crate then sounds good to me.